src/modules/hacker/HackerTest.cpp

```diff
@@ -0,0 +1,71 @@
+#include "gtest/gtest.h"
+#include "faker-cxx/Hacker.h"
+
+#include <algorithm>
+#include <string>
+#include <iostream>
+
+#include "data/Abbreviations.h"
+#include "data/Adjectives.h"
+#include "data/Ingverbs.h"
+#include "data/Nouns.h"
+#include "data/Verbs.h"
+#include "data/Phrases.h"
+
+// testing Hacker module, test like Person module tests
+// testing phrases will be different, test that each phrase will have a random word from each word list
+
+using namespace ::testing;
+using namespace faker;
+
+class HackerTest : public Test {
+public:
+};
+
+TEST_F(HackerTest, shouldGenerateAbbreviation) {
+    std::string generatedAbbreviation = Hacker::abbreviation();
+
+    ASSERT_TRUE(std::ranges::any_of(abbreviations, [generatedAbbreviation](const std::string& abbreviation) { return abbreviation == generatedAbbreviation; }));
+}
+
+TEST_F(HackerTest, shouldGenerateAdjective) {
+    std::string generatedAdjective = Hacker::adjective();
+
+    ASSERT_TRUE(std::ranges::any_of(adjectives, [generatedAdjective](const std::string& adjective) { return adjective == generatedAdjective; }));
+}
+
+TEST_F(HackerTest, shouldGenerateNoun) {
+    std::string generatedNoun = Hacker::noun();
+
+    ASSERT_TRUE(std::ranges::any_of(nouns, [generatedNoun](const std::string& noun) { return noun == generatedNoun; }));
+}
+
+TEST_F(HackerTest, shouldGenerateVerb) {
+    std::string generatedVerb = Hacker::verb();
+
+    ASSERT_TRUE(std::ranges::any_of(verbs, [generatedVerb](const std::string& verb) { return verb == generatedVerb; }));
+}
+
+TEST_F(HackerTest, shouldGenerateIngverb) {
+    std::string generatedIngverb = Hacker::ingverb();
+
+    ASSERT_TRUE(std::ranges::any_of(ingverbs, [generatedIngverb](const std::string& ingverb) { return ingverb == generatedIngverb; }));
+}
+
+TEST_F(HackerTest, shouldGeneratePhrase) {
```

Comment:
consider checking for common words like maybe all of the phrases have noun so lets assert it etc

Reply:
Alright, I'll try to work on that. This test was all I could come up with for the time being. I did think of that but I saw that some phrases didnt have ingverbs some did etc.

Reply:
yeah I would only assert for common parts and dont care about other parts, it will be hard to write some better test here